### PR TITLE
net/haproxy: release 4.0 [only for 23.1]

### PR DESCRIPTION
**Pull request by fraenki**
see plugin changelog for details

requires https://github.com/opnsense/tools/pull/332
fixes https://github.com/opnsense/plugins/issues/3026

### CODE REVIEW

**Comment by fraenki** [comment]
@fichtner This needs to be merged in-time for 23.1, but should not be shipped in any 22.7.x release. Feel free to merge whenever you want to start the 23.1 builds. I'll keep it in "draft" state for the time being, because I want to perform more tests next week.

**Comment by fraenki** [comment]
Further testing revealed no issues. :)

However, on one firewall a restart of HAProxy seems to take longer now. This may be related to the "seamless reloads" option, but it does not cause any other issue.

**Comment by fraenki** [comment]
@fichtner This is finally ready for OPNsense 23.1. Should I merge it now, or do you want to merge it when you start the 23.1 builds? Thanks :)

File: net/haproxy/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PLUGIN_NAME=		haproxy
-PLUGIN_VERSION=		3.12
+PLUGIN_VERSION=		4.0
 PLUGIN_COMMENT=		Reliable, high performance TCP/HTTP load balancer
-PLUGIN_DEPENDS=		haproxy24
+PLUGIN_DEPENDS=		haproxy
```
Comment (fichtner) [inline net/haproxy/Makefile:4]: I suppose we need the tools.git support for that? :)

**Comment by fichtner** [comment]
@fraenki feel free to merge when the required tools.git PR is open. I'll make sure to push it to stable/23.1 and not update stable/22.7 anymore.

**Review by fichtner — APPROVED** [review]

**Comment by fichtner** [comment]
BTW: nice work! 👍

File: net/haproxy/Makefile (hunk 1 above)
Comment (fraenki) [reply net/haproxy/Makefile:4]: PR https://github.com/opnsense/tools/pull/332 :)

Comment (fichtner) [reply net/haproxy/Makefile:4]: all done, thanks!

**Pull request merged**